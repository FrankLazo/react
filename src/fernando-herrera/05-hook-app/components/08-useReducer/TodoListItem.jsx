const TodoListItem = ({ todo, index, handleToggle, handleDelete }) =>
{
	return (
		<li className="columns has-not-bullet m-0">
			<div
				className={`column is-three-fifths has-cursor-pointer ${ todo.done && 'has-text-grey-dark' }`}
				onClick={ () => handleToggle(todo.id) }
				>
				{ index + 1 }. { todo.description }
			</div>

			<div className="column">
				<button
					className={`button ${ todo.done ? 'is-black' : 'is-danger'} mr-2`}
					>Edit</button>
				<button className={`button ${ todo.done ? 'is-black' : 'is-danger'} is-outlined`}
					onClick={ () => handleDelete(todo.id) }
					>Delete</button>
			</div>
		</li>
	);
};

export default TodoListItem;
