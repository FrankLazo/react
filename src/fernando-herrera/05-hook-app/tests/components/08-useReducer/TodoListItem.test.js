import { shallow } from 'enzyme';
import TodoListItem from '../../../components/08-useReducer/TodoListItem';
import demoTodos from '../../fixtures/demoTodos';

describe('<TodoListItem /> tests', () => {
	const handleDelete = jest.fn();
	const handleToggle = jest.fn();

	const wrapper = shallow(
		<TodoListItem
			todo={ demoTodos[0] }
			index={ 0 }
			handleToggle={ handleToggle }
			handleDelete={ handleDelete }
			/>
	);

	test('should show the right component', () => {
		expect(wrapper).toMatchSnapshot();
	});

	test('should call handleDelete function', () => {
		const deleteButton = wrapper.find('button').at(1);
		deleteButton.simulate('click');

		expect(handleDelete).toHaveBeenCalledWith( demoTodos[0].id );
	});

	test('should call handleDelete function', () => {
		const toggleButton = wrapper.find('div').at(0);
		toggleButton.simulate('click');

		expect(handleToggle).toHaveBeenCalledWith( demoTodos[0].id );
	});

	test('should show the right component content', () => {
		const content = `1. ${ demoTodos[0].description }`

		expect(wrapper.find('div').at(0).text().trim()).toBe(content);
	});

	test('should show the complete class in div', () => {
		expect(wrapper.find('div').at(0).hasClass('has-text-grey-dark')).toBe(true);
	});
});
