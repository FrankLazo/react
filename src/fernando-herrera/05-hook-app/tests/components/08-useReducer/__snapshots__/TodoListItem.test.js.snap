// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoListItem /> tests should show the right component 1`] = `
<li
  className="columns has-not-bullet m-0"
>
  <div
    className="column is-three-fifths has-cursor-pointer has-text-grey-dark"
    onClick={[Function]}
  >
    1
    . 
    Learn React
  </div>
  <div
    className="column"
  >
    <button
      className="button is-black mr-2"
    >
      Edit
    </button>
    <button
      className="button is-black is-outlined"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</li>
`;
